--- a/modules/nf-core/ctatsplicing/startocancerintrons/main.nf
+++ b/modules/nf-core/ctatsplicing/startocancerintrons/main.nf
@@ -53,10 +53,10 @@
     def create_igv_files = args.contains("--vis") ? "touch ${prefix}.introns.for_IGV.bed && touch ${prefix}.ctat-splicing.igv.html && touch ${prefix}.igv.tracks" : ""
     """
     ${create_igv_files}
-    touch ${prefix}.cancer_intron_reads.sorted.bam
-    touch ${prefix}.cancer_intron_reads.sorted.bam.bai
-    touch ${prefix}.gene_reads.sorted.sifted.bam
-    touch ${prefix}.gene_reads.sorted.sifted.bam.bai
+    echo '' | gzip > ${prefix}.cancer_intron_reads.sorted.bam
+    echo '' | gzip > ${prefix}.cancer_intron_reads.sorted.bam.bai
+    echo '' | gzip > ${prefix}.gene_reads.sorted.sifted.bam
+    echo '' | gzip > ${prefix}.gene_reads.sorted.sifted.bam.bai
     touch ${prefix}.cancer.introns
     touch ${prefix}.cancer.introns.prelim
     touch ${prefix}.introns
