--- a/modules/nf-core/disambiguate/main.nf
+++ b/modules/nf-core/disambiguate/main.nf
@@ -36,10 +36,10 @@
     stub:
     def prefix = task.ext.prefix ?: "${meta.id}"
     """
-    touch ${prefix}.disambiguatedSpeciesA.bam
-    touch ${prefix}.disambiguatedSpeciesB.bam
-    touch ${prefix}.ambiguousSpeciesA.bam
-    touch ${prefix}.ambiguousSpeciesB.bam
+    echo '' | gzip > ${prefix}.disambiguatedSpeciesA.bam
+    echo '' | gzip > ${prefix}.disambiguatedSpeciesB.bam
+    echo '' | gzip > ${prefix}.ambiguousSpeciesA.bam
+    echo '' | gzip > ${prefix}.ambiguousSpeciesB.bam
     touch ${prefix}_summary.txt
     """
 }
