--- a/modules/nf-core/isoseq/cluster/main.nf
+++ b/modules/nf-core/isoseq/cluster/main.nf
@@ -46,8 +46,8 @@
     stub:
     def prefix = task.ext.prefix ?: "${meta.id}"
     """
-    touch ${prefix}.transcripts.bam
-    touch ${prefix}.transcripts.bam.pbi
+    echo '' | gzip > ${prefix}.transcripts.bam
+    echo '' | gzip > ${prefix}.transcripts.bam.pbi
     touch ${prefix}.transcripts.cluster
     touch ${prefix}.transcripts.cluster_report.csv
     touch ${prefix}.transcripts.transcriptset.xml
