--- a/modules/nf-core/paraphase/main.nf
+++ b/modules/nf-core/paraphase/main.nf
@@ -66,10 +66,10 @@
     mkdir ${prefix}_paraphase_vcfs
 
     touch ${prefix}.paraphase.json
-    touch ${prefix}.paraphase.bam
-    touch ${prefix}.paraphase.bam.bai
+    echo '' | gzip > ${prefix}.paraphase.bam
+    echo '' | gzip > ${prefix}.paraphase.bam.bai
     echo '' | gzip > ${prefix}_paraphase_vcfs/${prefix}_stub.vcf.gz
-    touch ${prefix}_paraphase_vcfs/${prefix}_stub.vcf.gz.${index}
+    echo '' | gzip > ${prefix}_paraphase_vcfs/${prefix}_stub.vcf.gz.${index}
     """
 }
 
