--- a/modules/nf-core/pbccs/main.nf
+++ b/modules/nf-core/pbccs/main.nf
@@ -46,8 +46,8 @@
     stub:
     def prefix = task.ext.prefix ?: "${meta.id}"
     """
-    touch ${prefix}.chunk1.bam
-    touch ${prefix}.chunk1.bam.pbi
+    echo '' | gzip > ${prefix}.chunk1.bam
+    echo '' | gzip > ${prefix}.chunk1.bam.pbi
     touch ${prefix}.report.txt
     touch ${prefix}.report.json
     echo | gzip > ${prefix}.metrics.json.gz
