--- a/modules/nf-core/pmdtools/filter/main.nf
+++ b/modules/nf-core/pmdtools/filter/main.nf
@@ -58,7 +58,7 @@
     if ("$bam" == "${prefix}.bam") error "[pmdtools/filter] Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"
     //threshold and header flags activate filtering function of pmdtools
     """
-    touch ${prefix}.bam
+    echo '' | gzip > ${prefix}.bam
 
     cat <<-END_VERSIONS > versions.yml
     "${task.process}":
