--- a/modules/nf-core/samtools/ampliconclip/main.nf
+++ b/modules/nf-core/samtools/ampliconclip/main.nf
@@ -46,7 +46,7 @@
     stub:
 
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def rejects = save_cliprejects ? "touch ${prefix}.cliprejects.bam" : ""
+    def rejects = save_cliprejects ? "echo '' | gzip > ${prefix}.cliprejects.bam" : ""
     def stats = save_clipstats ? "touch ${prefix}.clipstats.txt" : ""
 
     if ("${bam}" == "${prefix}.bam") {
