diff --git a/modules/nf-core/deepvariant/main.nf b/modules/nf-core/deepvariant/main.nf
index 85c4f71..a280b0e 100644
--- a/modules/nf-core/deepvariant/main.nf
+++ b/modules/nf-core/deepvariant/main.nf
@@ -36,5 +36,16 @@ The processing stages used by the subworkflow are implemented as module subcomma
 """
     prefix = task.ext.prefix ?: "${meta.id}"
     assert false: deprecation_message
+    stub:
+    """
+    echo | gzip > ${prefix}.vcf.gz
+    touch ${prefix}.vcf.gz.tbi
+    echo | gzip > ${prefix}.g.vcf.gz
+    touch ${prefix}.g.vcf.gz.tbi
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            deepvariant: \$(echo "stub")
+        END_VERSIONS
+    """
 
 }
diff --git a/modules/nf-core/ectyper/main.nf b/modules/nf-core/ectyper/main.nf
index 3245e8a..b031199 100644
--- a/modules/nf-core/ectyper/main.nf
+++ b/modules/nf-core/ectyper/main.nf
@@ -42,4 +42,15 @@ process ECTYPER {
         ectyper: \$(echo \$(ectyper --version 2>&1)  | sed 's/.*ectyper //; s/ .*\$//')
     END_VERSIONS
     """
+    stub:
+    """
+    touch ectyper_stub.txt
+    touch ectyper_stub.log
+    touch ectyper_stub.tsv
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            ectyper: \$(echo \$(ectyper --version 2>&1)  | sed 's/.*ectyper //; s/ .*\$//')
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/emmtyper/main.nf b/modules/nf-core/emmtyper/main.nf
index 53f7d7f..5cf9830 100644
--- a/modules/nf-core/emmtyper/main.nf
+++ b/modules/nf-core/emmtyper/main.nf
@@ -31,4 +31,13 @@ process EMMTYPER {
         emmtyper: \$( echo \$(emmtyper --version 2>&1) | sed 's/^.*emmtyper v//' )
     END_VERSIONS
     """
+    stub:
+    """
+    touch emmtyper_stub.tsv
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            emmtyper: \$( echo \$(emmtyper --version 2>&1) | sed 's/^.*emmtyper v//' )
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/fastqscan/main.nf b/modules/nf-core/fastqscan/main.nf
index ac5c13f..cbaa637 100644
--- a/modules/nf-core/fastqscan/main.nf
+++ b/modules/nf-core/fastqscan/main.nf
@@ -30,4 +30,13 @@ process FASTQSCAN {
         fastqscan: \$( echo \$(fastq-scan -v 2>&1) | sed 's/^.*fastq-scan //' )
     END_VERSIONS
     """
+    stub:
+    """
+    touch fastqscan_stub.json
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            fastqscan: \$( echo \$(fastq-scan -v 2>&1) | sed 's/^.*fastq-scan //' )
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/fasttree/main.nf b/modules/nf-core/fasttree/main.nf
index d65139e..2a74b98 100644
--- a/modules/nf-core/fasttree/main.nf
+++ b/modules/nf-core/fasttree/main.nf
@@ -30,4 +30,13 @@ process FASTTREE {
         fasttree: \$(fasttree -help 2>&1 | head -1  | sed 's/^FastTree \\([0-9\\.]*\\) .*\$/\\1/')
     END_VERSIONS
     """
+    stub:
+    """
+    touch fasttree_stub.tre
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            fasttree: \$(fasttree -help 2>&1 | head -1  | sed 's/^FastTree \\([0-9\\.]*\\) .*\$/\\1/')
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/ffq/main.nf b/modules/nf-core/ffq/main.nf
index db1db12..838cdb4 100644
--- a/modules/nf-core/ffq/main.nf
+++ b/modules/nf-core/ffq/main.nf
@@ -33,4 +33,13 @@ process FFQ {
         ffq: \$(echo \$(ffq --help 2>&1) | sed 's/^.*ffq //; s/: A command.*\$//' )
     END_VERSIONS
     """
+    stub:
+    """
+    touch ffq_stub.json
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            ffq: \$(echo \$(ffq --help 2>&1) | sed 's/^.*ffq //; s/: A command.*\$//' )
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/filtlong/main.nf b/modules/nf-core/filtlong/main.nf
index 0653974..a29bc56 100644
--- a/modules/nf-core/filtlong/main.nf
+++ b/modules/nf-core/filtlong/main.nf
@@ -31,4 +31,14 @@ process FILTLONG {
         2>| >(tee ${prefix}.log >&2) \\
         | gzip -n > ${prefix}.fastq.gz
     """
+    stub:
+    """
+    touch filtlong_stub.log
+    echo | gzip > filtlong_stub.fastq.gz
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            filtlong: \$(echo "stub")
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/genrich/main.nf b/modules/nf-core/genrich/main.nf
index 7a712ec..4a67707 100644
--- a/modules/nf-core/genrich/main.nf
+++ b/modules/nf-core/genrich/main.nf
@@ -51,4 +51,17 @@ process GENRICH {
         genrich: \$(echo \$(Genrich --version 2>&1) | sed 's/^Genrich, version //; s/ .*\$//')
     END_VERSIONS
     """
+    stub:
+    """
+    touch genrich_stub.pileup.bedGraph
+    touch genrich_stub.narrowPeak
+    touch genrich_stub.duplicates.txt
+    touch genrich_stub.pvalues.bedGraph
+    touch genrich_stub.intervals.bed
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            genrich: \$(echo \$(Genrich --version 2>&1) | sed 's/^Genrich, version //; s/ .*\$//')
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/islandpath/main.nf b/modules/nf-core/islandpath/main.nf
index b3fdcd9..2bff040 100644
--- a/modules/nf-core/islandpath/main.nf
+++ b/modules/nf-core/islandpath/main.nf
@@ -34,4 +34,14 @@ process ISLANDPATH {
         islandpath: $VERSION
     END_VERSIONS
     """
+    stub:
+    """
+    touch islandpath_stub.gff
+    touch Dimob.log
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            islandpath: $VERSION
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/kofamscan/main.nf b/modules/nf-core/kofamscan/main.nf
index 4360fd3..a5a0fb6 100644
--- a/modules/nf-core/kofamscan/main.nf
+++ b/modules/nf-core/kofamscan/main.nf
@@ -39,4 +39,14 @@ process KOFAMSCAN {
         kofamscan: \$(echo \$(exec_annotation --version 2>&1) | sed 's/^.*exec_annotation //;')
     END_VERSIONS
     """
+    stub:
+    """
+    touch kofamscan_stub.txt
+    touch kofamscan_stub.tsv
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            kofamscan: \$(echo \$(exec_annotation --version 2>&1) | sed 's/^.*exec_annotation //;')
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/maltextract/main.nf b/modules/nf-core/maltextract/main.nf
index 386d64f..1ddce84 100644
--- a/modules/nf-core/maltextract/main.nf
+++ b/modules/nf-core/maltextract/main.nf
@@ -36,4 +36,13 @@ process MALTEXTRACT {
         maltextract: \$(MaltExtract --help | head -n 2 | tail -n 1 | sed 's/MaltExtract version//')
     END_VERSIONS
     """
+    stub:
+    """
+    touch results
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            maltextract: \$(MaltExtract --help | head -n 2 | tail -n 1 | sed 's/MaltExtract version//')
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/medaka/main.nf b/modules/nf-core/medaka/main.nf
index d741835..d10c880 100644
--- a/modules/nf-core/medaka/main.nf
+++ b/modules/nf-core/medaka/main.nf
@@ -37,4 +37,13 @@ process MEDAKA {
         medaka: \$( medaka --version 2>&1 | sed 's/medaka //g' )
     END_VERSIONS
     """
+    stub:
+    """
+    echo | gzip > medaka_stub.fa.gz
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            medaka: \$( medaka --version 2>&1 | sed 's/medaka //g' )
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/mlst/main.nf b/modules/nf-core/mlst/main.nf
index 19d9f6c..0643e0b 100644
--- a/modules/nf-core/mlst/main.nf
+++ b/modules/nf-core/mlst/main.nf
@@ -32,5 +32,13 @@ process MLST {
         mlst: \$( echo \$(mlst --version 2>&1) | sed 's/mlst //' )
     END_VERSIONS
     """
+    stub:
+    """
+    touch mlst_stub.tsv
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            mlst: \$( echo \$(mlst --version 2>&1) | sed 's/mlst //' )
+        END_VERSIONS
+    """
 
 }
diff --git a/modules/nf-core/mygene/main.nf b/modules/nf-core/mygene/main.nf
index 70141d1..199aa4f 100644
--- a/modules/nf-core/mygene/main.nf
+++ b/modules/nf-core/mygene/main.nf
@@ -20,4 +20,14 @@ process MYGENE {
 
     script:
     template "mygene.py"
+    stub:
+    """
+    touch mygene_stub.gmt
+    touch mygene_stub.tsv
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            mygene: \$(echo "stub")
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/ncbigenomedownload/main.nf b/modules/nf-core/ncbigenomedownload/main.nf
index 8ef6513..c99cbd0 100644
--- a/modules/nf-core/ncbigenomedownload/main.nf
+++ b/modules/nf-core/ncbigenomedownload/main.nf
@@ -47,4 +47,25 @@ process NCBIGENOMEDOWNLOAD {
         $groups
 
     """
+    stub:
+    """
+    touch ncbigenomedownload_stub_assembly_report.txt
+    touch ncbigenomedownload_stub_assembly_stats.txt
+    echo | gzip > ncbigenomedownload_stub_wgsmaster.gbff.gz
+    echo | gzip > ncbigenomedownload_stub_rna_from_genomic.fna.gz
+    echo | gzip > ncbigenomedownload_stub_feature_table.txt.gz
+    echo | gzip > ncbigenomedownload_stub_rna.fna.gz
+    echo | gzip > ncbigenomedownload_stub_genomic.fna.gz
+    echo | gzip > ncbigenomedownload_stub_cds_from_genomic.fna.gz
+    echo | gzip > ncbigenomedownload_stub_protein.gpff.gz
+    echo | gzip > ncbigenomedownload_stub_genomic.gff.gz
+    echo | gzip > ncbigenomedownload_stub_genomic.gbff.gz
+    echo | gzip > ncbigenomedownload_stub_rm.out.gz
+    echo | gzip > ncbigenomedownload_stub_protein.faa.gz
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            ncbigenomedownload: \$(echo "stub")
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/optitype/main.nf b/modules/nf-core/optitype/main.nf
index 0508e88..3d726ad 100644
--- a/modules/nf-core/optitype/main.nf
+++ b/modules/nf-core/optitype/main.nf
@@ -47,4 +47,14 @@ process OPTITYPE {
         optitype: \$(cat \$(which OptiTypePipeline.py) | grep -e "Version:" | sed -e "s/Version: //g")
     END_VERSIONS
     """
+    stub:
+    """
+    touch ${prefix}/optitype_stub.pdf
+    touch ${prefix}/optitype_stub.tsv
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            optitype: \$(cat \$(which OptiTypePipeline.py) | grep -e "Version:" | sed -e "s/Version: //g")
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/pairix/main.nf b/modules/nf-core/pairix/main.nf
index f0c2017..69654de 100644
--- a/modules/nf-core/pairix/main.nf
+++ b/modules/nf-core/pairix/main.nf
@@ -29,4 +29,13 @@ process PAIRIX {
         pairix: \$(echo \$(pairix --help 2>&1) | sed 's/^.*Version: //; s/Usage.*\$//')
     END_VERSIONS
     """
+    stub:
+    """
+    touch pairix_stub.px2
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            pairix: \$(echo \$(pairix --help 2>&1) | sed 's/^.*Version: //; s/Usage.*\$//')
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/plasmidid/main.nf b/modules/nf-core/plasmidid/main.nf
index 4420478..5de448c 100644
--- a/modules/nf-core/plasmidid/main.nf
+++ b/modules/nf-core/plasmidid/main.nf
@@ -42,4 +42,20 @@ process PLASMIDID {
         plasmidid: \$(echo \$(plasmidID --version 2>&1))
     END_VERSIONS
     """
+    stub:
+    """
+    touch ${prefix}/kmer/
+    touch ${prefix}/images/
+    touch ${prefix}/plasmidid_stubfinal_results.html
+    touch ${prefix}/logs/
+    touch ${prefix}/plasmidid_stubfinal_results.tab
+    touch ${prefix}/database/
+    touch ${prefix}/fasta_files/
+    touch ${prefix}/data/
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            plasmidid: \$(echo \$(plasmidID --version 2>&1))
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/prinseqplusplus/main.nf b/modules/nf-core/prinseqplusplus/main.nf
index 403a51b..681c7d2 100644
--- a/modules/nf-core/prinseqplusplus/main.nf
+++ b/modules/nf-core/prinseqplusplus/main.nf
@@ -58,4 +58,16 @@ process PRINSEQPLUSPLUS {
         END_VERSIONS
         """
     }
+    stub:
+    """
+    echo | gzip > prinseqplusplus_stub_good_outprinseqplusplus_stub.fastq.gz
+    echo | gzip > prinseqplusplus_stub_bad_outprinseqplusplus_stub.fastq.gz
+    echo | gzip > prinseqplusplus_stub_single_outprinseqplusplus_stub.fastq.gz
+    touch prinseqplusplus_stub.log
+    cat <<-END_VERSIONS > versions.yml
+            "${task.process}":
+                prinseqplusplus: \$(echo \$(prinseq++ --version | cut -f 2 -d ' ' ))
+            END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/pychopper/main.nf b/modules/nf-core/pychopper/main.nf
index 93fe0f2..39b7a34 100644
--- a/modules/nf-core/pychopper/main.nf
+++ b/modules/nf-core/pychopper/main.nf
@@ -45,7 +45,7 @@ process PYCHOPPER {
 
     cat <<-END_VERSIONS > versions.yml
     "${task.process}":
-        pychopper: 2.7.10 (hard coded- check container used for this module)
+        pychopper: $PYCHOPPER_VERSION (hard coded- check container used for this module)
     END_VERSIONS
     """
 }
diff --git a/modules/nf-core/pycoqc/main.nf b/modules/nf-core/pycoqc/main.nf
index e748aac..82d5f22 100644
--- a/modules/nf-core/pycoqc/main.nf
+++ b/modules/nf-core/pycoqc/main.nf
@@ -33,4 +33,14 @@ process PYCOQC {
         pycoqc: \$(pycoQC --version 2>&1 | sed 's/^.*pycoQC v//; s/ .*\$//')
     END_VERSIONS
     """
+    stub:
+    """
+    touch pycoqc_stub.html
+    touch pycoqc_stub.json
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            pycoqc: \$(pycoQC --version 2>&1 | sed 's/^.*pycoQC v//; s/ .*\$//')
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/racon/main.nf b/modules/nf-core/racon/main.nf
index dd9fe6b..ef48cc2 100644
--- a/modules/nf-core/racon/main.nf
+++ b/modules/nf-core/racon/main.nf
@@ -35,4 +35,13 @@ process RACON {
         racon: \$( racon --version 2>&1 | sed 's/^.*v//' )
     END_VERSIONS
     """
+    stub:
+    """
+    echo | gzip > racon_stub_assembly_consensus.fasta.gz
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            racon: \$( racon --version 2>&1 | sed 's/^.*v//' )
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/rapidnj/main.nf b/modules/nf-core/rapidnj/main.nf
index e5df6ee..0987af8 100644
--- a/modules/nf-core/rapidnj/main.nf
+++ b/modules/nf-core/rapidnj/main.nf
@@ -38,4 +38,15 @@ process RAPIDNJ {
         biopython: \$(python -c "import Bio; print(Bio.__version__)")
     END_VERSIONS
     """
+    stub:
+    """
+    touch rapidnj_stub.sth
+    touch rapidnj_stub.tre
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            rapidnj: $VERSION
+            biopython: \$(python -c "import Bio; print(Bio.__version__)")
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/rasusa/main.nf b/modules/nf-core/rasusa/main.nf
index a1d36de..6dccd67 100644
--- a/modules/nf-core/rasusa/main.nf
+++ b/modules/nf-core/rasusa/main.nf
@@ -30,4 +30,13 @@ process RASUSA {
         --input $reads \\
         $output
     """
+    stub:
+    """
+    echo | gzip > rasusa_stub.fastq.gz
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            rasusa: \$(echo "stub")
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/raven/main.nf b/modules/nf-core/raven/main.nf
index f01a9ae..22d8a78 100644
--- a/modules/nf-core/raven/main.nf
+++ b/modules/nf-core/raven/main.nf
@@ -33,4 +33,14 @@ process RAVEN {
     # compress assembly graph
     gzip -c ${prefix}.gfa > ${prefix}.gfa.gz
     """
+    stub:
+    """
+    echo | gzip > raven_stub.gfa.gz
+    echo | gzip > raven_stub.fasta.gz
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            raven: \$(echo "stub")
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/salsa2/main.nf b/modules/nf-core/salsa2/main.nf
index d435d34..fd76c1f 100644
--- a/modules/nf-core/salsa2/main.nf
+++ b/modules/nf-core/salsa2/main.nf
@@ -49,4 +49,15 @@ process SALSA2 {
         SALSA2: $VERSION
     END_VERSIONS
     """
+    stub:
+    """
+    touch salsa2_stub_scaffolds_FINAL.agp
+    touch salsa2_stub/salsa2_stubscaffolds_FINAL.original-coordinates.agp
+    touch salsa2_stub_scaffolds_FINAL.fasta
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            SALSA2: $VERSION
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/scoary/main.nf b/modules/nf-core/scoary/main.nf
index df3f713..85b6524 100644
--- a/modules/nf-core/scoary/main.nf
+++ b/modules/nf-core/scoary/main.nf
@@ -35,4 +35,13 @@ process SCOARY {
         scoary: \$( scoary --version 2>&1 )
     END_VERSIONS
     """
+    stub:
+    """
+    touch scoary_stub.csv
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            scoary: \$( scoary --version 2>&1 )
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/seqsero2/main.nf b/modules/nf-core/seqsero2/main.nf
index 599fce5..0f7549a 100644
--- a/modules/nf-core/seqsero2/main.nf
+++ b/modules/nf-core/seqsero2/main.nf
@@ -35,4 +35,15 @@ process SEQSERO2 {
         seqsero2: \$( echo \$( SeqSero2_package.py --version 2>&1) | sed 's/^.*SeqSero2_package.py //' )
     END_VERSIONS
     """
+    stub:
+    """
+    touch results/seqsero2_stub_log.txt
+    touch results/seqsero2_stub_result.tsv
+    touch results/seqsero2_stub_result.txt
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            seqsero2: \$( echo \$( SeqSero2_package.py --version 2>&1) | sed 's/^.*SeqSero2_package.py //' )
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/shasta/main.nf b/modules/nf-core/shasta/main.nf
index b55df9f..5dabff2 100644
--- a/modules/nf-core/shasta/main.nf
+++ b/modules/nf-core/shasta/main.nf
@@ -46,4 +46,15 @@ process SHASTA {
         shasta: \$(shasta --version | head -n 1 | cut -f 3 -d " ")
     END_VERSIONS
     """
+    stub:
+    """
+    echo | gzip > shasta_stub_Assembly.gfa.gz
+    echo | gzip > shasta_stub_Assembly.fasta.gz
+    touch ShastaRun/
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            shasta: \$(shasta --version | head -n 1 | cut -f 3 -d " ")
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/shasum/main.nf b/modules/nf-core/shasum/main.nf
index 157529c..10be535 100644
--- a/modules/nf-core/shasum/main.nf
+++ b/modules/nf-core/shasum/main.nf
@@ -30,4 +30,13 @@ process SHASUM {
         sha256sum: \$(echo \$(sha256sum --version 2>&1 | head -n 1| sed 's/^.*) //;' ))
     END_VERSIONS
     """
+    stub:
+    """
+    touch shasum_stub.sha256
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            sha256sum: \$(echo \$(sha256sum --version 2>&1 | head -n 1| sed 's/^.*) //;' ))
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/shovill/main.nf b/modules/nf-core/shovill/main.nf
index c54c68f..48fdcc6 100644
--- a/modules/nf-core/shovill/main.nf
+++ b/modules/nf-core/shovill/main.nf
@@ -34,4 +34,17 @@ process SHOVILL {
         --outdir ./ \\
         --force
     """
+    stub:
+    """
+    touch contigs.{fastg,gfa,LastGraph}
+    touch shovill.log
+    touch {skesa,spades,megahit,velvet}.fasta
+    touch shovill.corrections
+    touch contigs.fa
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            shovill: \$(echo "stub")
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/sickle/main.nf b/modules/nf-core/sickle/main.nf
index 26f3860..9e647a5 100644
--- a/modules/nf-core/sickle/main.nf
+++ b/modules/nf-core/sickle/main.nf
@@ -61,4 +61,16 @@ process SICKLE {
     END_VERSIONS
     """
     }
+    stub:
+    """
+    echo | gzip > ${prefix}.se.trimmed.fastq.gz
+    echo | gzip > ${prefix}.singleton.trimmed.fastq.gz
+    echo | gzip > ${prefix}.pe{1,2}.trimmed.fastq.gz
+    touch sickle_stub.log
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            sickle: \$(sickle --version|awk 'NR==1{print \$3}')
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/sistr/main.nf b/modules/nf-core/sistr/main.nf
index efa2e1e..e63785f 100644
--- a/modules/nf-core/sistr/main.nf
+++ b/modules/nf-core/sistr/main.nf
@@ -46,4 +46,16 @@ process SISTR {
         sistr: \$(echo \$(sistr --version 2>&1) | sed 's/^.*sistr_cmd //; s/ .*\$//' )
     END_VERSIONS
     """
+    stub:
+    """
+    touch sistr_stub-allele.json
+    touch sistr_stub-cgmlst.csv
+    touch sistr_stub-allele.fasta
+    touch sistr_stub.tab
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            sistr: \$(echo \$(sistr --version 2>&1) | sed 's/^.*sistr_cmd //; s/ .*\$//' )
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/slimfastq/main.nf b/modules/nf-core/slimfastq/main.nf
index bbdf159..a6ba63a 100644
--- a/modules/nf-core/slimfastq/main.nf
+++ b/modules/nf-core/slimfastq/main.nf
@@ -49,4 +49,13 @@ process SLIMFASTQ {
         END_VERSIONS
         """
     }
+    stub:
+    """
+    touch slimfastq_stub.sfq
+    cat <<-END_VERSIONS > versions.yml
+            "${task.process}":
+                slimfastq: ${VERSION}
+            END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/smoothxg/main.nf b/modules/nf-core/smoothxg/main.nf
index 8a32e60..2bf9e63 100644
--- a/modules/nf-core/smoothxg/main.nf
+++ b/modules/nf-core/smoothxg/main.nf
@@ -33,4 +33,14 @@ process SMOOTHXG {
         smoothxg: \$(smoothxg --version 2>&1 | cut -f 1 -d '-' | cut -f 2 -d 'v')
     END_VERSIONS
     """
+    stub:
+    """
+    touch smoothxg_stub.maf
+    touch smoothxg_stubsmoothxg.gfa
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            smoothxg: \$(smoothxg --version 2>&1 | cut -f 1 -d '-' | cut -f 2 -d 'v')
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/snpdists/main.nf b/modules/nf-core/snpdists/main.nf
index 8884a38..131e83e 100644
--- a/modules/nf-core/snpdists/main.nf
+++ b/modules/nf-core/snpdists/main.nf
@@ -30,4 +30,13 @@ process SNPDISTS {
         snpdists: \$(snp-dists -v 2>&1 | sed 's/snp-dists //;')
     END_VERSIONS
     """
+    stub:
+    """
+    touch snpdists_stub.tsv
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            snpdists: \$(snp-dists -v 2>&1 | sed 's/snp-dists //;')
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/spatyper/main.nf b/modules/nf-core/spatyper/main.nf
index 641dbbc..8707adb 100644
--- a/modules/nf-core/spatyper/main.nf
+++ b/modules/nf-core/spatyper/main.nf
@@ -39,5 +39,10 @@ process SPATYPER {
     def prefix = task.ext.prefix ?: "${meta.id}"
     """
     touch ${prefix}.tsv
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        spatyper: \$( echo \$(spaTyper --version 2>&1) | sed 's/^.*spaTyper //' )
+    END_VERSIONS
     """
 }
diff --git a/modules/nf-core/ssuissero/main.nf b/modules/nf-core/ssuissero/main.nf
index 7abc7a8..e1b1008 100644
--- a/modules/nf-core/ssuissero/main.nf
+++ b/modules/nf-core/ssuissero/main.nf
@@ -40,4 +40,13 @@ process SSUISSERO {
         ssuissero: $VERSION
     END_VERSIONS
     """
+    stub:
+    """
+    touch ssuissero_stub.tsv
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            ssuissero: $VERSION
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/staphopiasccmec/main.nf b/modules/nf-core/staphopiasccmec/main.nf
index 35427a3..e15d68d 100644
--- a/modules/nf-core/staphopiasccmec/main.nf
+++ b/modules/nf-core/staphopiasccmec/main.nf
@@ -28,4 +28,13 @@ process STAPHOPIASCCMEC {
         staphopiasccmec: \$(staphopia-sccmec --version 2>&1 | sed 's/^.*staphopia-sccmec //')
     END_VERSIONS
     """
+    stub:
+    """
+    touch staphopiasccmec_stub.tsv
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            staphopiasccmec: \$(staphopia-sccmec --version 2>&1 | sed 's/^.*staphopia-sccmec //')
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/tailfindr/main.nf b/modules/nf-core/tailfindr/main.nf
index 02e9fde..319d762 100644
--- a/modules/nf-core/tailfindr/main.nf
+++ b/modules/nf-core/tailfindr/main.nf
@@ -35,4 +35,14 @@ process TAILFINDR {
         ont-fast5-api: \$(pip show ont-fast5-api | grep Version | awk '{print \$2}')
     END_VERSIONS
     """
+    stub:
+    """
+    echo | gzip > tailfindr_stub.csv.gz
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            tailfindr: \$(Rscript -e "cat(paste(packageVersion('tailfindr'), collapse='.'))")
+            ont-fast5-api: \$(pip show ont-fast5-api | grep Version | awk '{print \$2}')
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/universc/main.nf b/modules/nf-core/universc/main.nf
index 0c08243..56591b1 100644
--- a/modules/nf-core/universc/main.nf
+++ b/modules/nf-core/universc/main.nf
@@ -96,7 +96,7 @@ process UNIVERSC {
     cat <<-END_VERSIONS > versions.yml
     "${task.process}":
         cellranger:  \$(echo \$(cellranger count --version 2>&1 | head -n 2 | tail -n 1 | sed 's/^.* //g' | sed 's/(//g' | sed 's/)//g' ))
-        universc:  \$(echo \$(bash /universc/launch_universc.sh --version | grep version | grep universc | sed 's/^.* //g' ))
+        universc:  \$(echo \$(bash /universc/launch_universc.sh --version | grep version | grep universc  | sed 's/^.* //g' ))
     END_VERSIONS
     """
 }
diff --git a/modules/nf-core/wfmash/main.nf b/modules/nf-core/wfmash/main.nf
index 4cb9b7a..0c3fb3d 100644
--- a/modules/nf-core/wfmash/main.nf
+++ b/modules/nf-core/wfmash/main.nf
@@ -41,4 +41,13 @@ process WFMASH {
         wfmash: \$(echo \$(wfmash --version 2>&1) | cut -f 1 -d '-' | cut -f 2 -d 'v')
     END_VERSIONS
     """
+    stub:
+    """
+    touch wfmash_stub.paf
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            wfmash: \$(echo \$(wfmash --version 2>&1) | cut -f 1 -d '-' | cut -f 2 -d 'v')
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/whamg/main.nf b/modules/nf-core/whamg/main.nf
index 47c4d70..867c953 100644
--- a/modules/nf-core/whamg/main.nf
+++ b/modules/nf-core/whamg/main.nf
@@ -42,4 +42,15 @@ process WHAMG {
         whamg: \$(echo \$(whamg 2>&1 | grep Version | sed 's/^Version: v//; s/-.*\$//' ))
     END_VERSIONS
     """
+    stub:
+    """
+    touch whamg_stub.vcf.gz.tbi
+    echo | gzip > whamg_stub.vcf.gz
+    touch whamg_stub.txt
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            whamg: \$(echo \$(whamg 2>&1 | grep Version | sed 's/^Version: v//; s/-.*\$//' ))
+        END_VERSIONS
+    """
+
 }
diff --git a/modules/nf-core/zip/main.nf b/modules/nf-core/zip/main.nf
index 35d3e0e..44dc41d 100644
--- a/modules/nf-core/zip/main.nf
+++ b/modules/nf-core/zip/main.nf
@@ -33,4 +33,13 @@ process ZIP {
         7za: \$(echo \$(7za --help) | sed 's/.*p7zip Version //; s/(.*//')
     END_VERSIONS
     """
+    stub:
+    """
+    touch ${prefix}.zip
+    cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            7za: \$(echo \$(7za --help) | sed 's/.*p7zip Version //; s/(.*//')
+        END_VERSIONS
+    """
+
 }
