diff --git a/modules/nf-core/fasttree/main.nf b/modules/nf-core/fasttree/main.nf
index d65139e6cd..2cb9e9c34c 100644
--- a/modules/nf-core/fasttree/main.nf
+++ b/modules/nf-core/fasttree/main.nf
@@ -30,4 +30,14 @@ process FASTTREE {
         fasttree: \$(fasttree -help 2>&1 | head -1  | sed 's/^FastTree \\([0-9\\.]*\\) .*\$/\\1/')
     END_VERSIONS
     """
+
+    stub:
+    """
+    touch fasttree_phylogeny.tre
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        fasttree: \$(fasttree -help 2>&1 | head -1  | sed 's/^FastTree \\([0-9\\.]*\\) .*\$/\\1/')
+    END_VERSIONS
+    """
 }
diff --git a/modules/nf-core/fasttree/tests/main.nf.test b/modules/nf-core/fasttree/tests/main.nf.test
index 1ae1ecae24..157fd869b1 100644
--- a/modules/nf-core/fasttree/tests/main.nf.test
+++ b/modules/nf-core/fasttree/tests/main.nf.test
@@ -27,5 +27,26 @@ nextflow_process {
             )
         }
     }
+    
+    test("test-fasttree - stub") {
+
+        options "-stub"
+
+        when {
+            process {
+                """
+                input[0] = [ file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/alignment/informative_sites.fas', checkIfExists: true) ]
+
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+    }
 
 }
diff --git a/modules/nf-core/rapidnj/main.nf b/modules/nf-core/rapidnj/main.nf
index e5df6eeb57..8e3a7c2aea 100644
--- a/modules/nf-core/rapidnj/main.nf
+++ b/modules/nf-core/rapidnj/main.nf
@@ -38,4 +38,17 @@ process RAPIDNJ {
         biopython: \$(python -c "import Bio; print(Bio.__version__)")
     END_VERSIONS
     """
+
+    stub:
+    def VERSION = '2.3.2' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    """
+    touch alignment.sth
+    touch rapidnj_phylogeny.tre
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        rapidnj: $VERSION
+        biopython: \$(python -c "import Bio; print(Bio.__version__)")
+    END_VERSIONS
+    """
 }
diff --git a/modules/nf-core/rapidnj/tests/main.nf.test b/modules/nf-core/rapidnj/tests/main.nf.test
index 8bd5bd7d0d..40e5ff8795 100644
--- a/modules/nf-core/rapidnj/tests/main.nf.test
+++ b/modules/nf-core/rapidnj/tests/main.nf.test
@@ -33,4 +33,30 @@ nextflow_process {
         }
     }
 
+    test("sarscov2-fasta - stub") {
+
+        options "-stub"
+
+        when {
+            process {
+                """
+                input[0] = [ file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/alignment/informative_sites.fas', checkIfExists: true) ]
+
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert process.out.phylogeny },
+                { assert snapshot(
+					process.out.stockholm_alignment,
+					process.out.versions
+					).match()
+				}
+            )
+        }
+    }
+
 }
