diff --git a/modules/nf-core/medaka/main.nf b/modules/nf-core/medaka/main.nf
index d741835..0586e4d 100644
--- a/modules/nf-core/medaka/main.nf
+++ b/modules/nf-core/medaka/main.nf
@@ -37,4 +37,15 @@ process MEDAKA {
         medaka: \$( medaka --version 2>&1 | sed 's/medaka //g' )
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    echo "" | gzip > ${prefix}.fa.gz
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        medaka: \$( medaka --version 2>&1 | sed 's/medaka //g' )
+    END_VERSIONS
+    """
 }
diff --git a/modules/nf-core/medaka/tests/main.nf.test b/modules/nf-core/medaka/tests/main.nf.test
index 948e398..4a7b454 100644
--- a/modules/nf-core/medaka/tests/main.nf.test
+++ b/modules/nf-core/medaka/tests/main.nf.test
@@ -34,4 +34,29 @@ nextflow_process {
 
     }
 
+    test("Medaka - stub") {
+
+        options "-stub"
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test', single_end:true ], // meta map
+                    file(params.modules_testdata_base_path + 'genomics/sarscov2/nanopore/fastq/test.fastq.gz', checkIfExists: true),
+                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+
+    }
+
 }
diff --git a/modules/nf-core/racon/main.nf b/modules/nf-core/racon/main.nf
index dd9fe6b..9d5a67a 100644
--- a/modules/nf-core/racon/main.nf
+++ b/modules/nf-core/racon/main.nf
@@ -35,4 +35,15 @@ process RACON {
         racon: \$( racon --version 2>&1 | sed 's/^.*v//' )
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    echo "" | gzip > ${prefix}_assembly_consensus.fasta.gz
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        racon: \$( racon --version 2>&1 | sed 's/^.*v//' )
+    END_VERSIONS
+    """
 }
diff --git a/modules/nf-core/racon/tests/main.nf.test b/modules/nf-core/racon/tests/main.nf.test
index 8ba02b6..c134443 100644
--- a/modules/nf-core/racon/tests/main.nf.test
+++ b/modules/nf-core/racon/tests/main.nf.test
@@ -31,4 +31,30 @@ nextflow_process {
         }
 
     }
+
+    test("single-end - bacteroides_fragilis - [fastq_gz, fna_gz, paf] - stub") {
+
+        options "-stub"
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test', single_end:true ], // meta map
+                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/nanopore/fastq/test.fastq.gz', checkIfExists: true),
+                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true),
+                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/genome/genome.paf', checkIfExists: true)
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll (
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+
+    }
 }
diff --git a/modules/nf-core/raven/main.nf b/modules/nf-core/raven/main.nf
index f01a9ae..485f6ef 100644
--- a/modules/nf-core/raven/main.nf
+++ b/modules/nf-core/raven/main.nf
@@ -33,4 +33,11 @@ process RAVEN {
     # compress assembly graph
     gzip -c ${prefix}.gfa > ${prefix}.gfa.gz
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    echo "" | gzip > ${prefix}.fasta.gz
+    echo "" | gzip > ${prefix}.gfa.gz
+    """
 }
diff --git a/modules/nf-core/raven/tests/main.nf.test b/modules/nf-core/raven/tests/main.nf.test
index cddfa41..188af9c 100644
--- a/modules/nf-core/raven/tests/main.nf.test
+++ b/modules/nf-core/raven/tests/main.nf.test
@@ -31,4 +31,28 @@ nextflow_process {
         }
     }
 
+    test("test-raven - stub") {
+
+        options "-stub"
+
+        when {
+            process {
+                """
+                input[0] = [
+				    [ id:'test', single_end:false ], // meta map
+				    [ file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/nanopore/fastq/test.fastq.gz', checkIfExists: true) ]
+				]
+
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+    }
+
 }
diff --git a/modules/nf-core/salsa2/main.nf b/modules/nf-core/salsa2/main.nf
index d435d34..665f2b8 100644
--- a/modules/nf-core/salsa2/main.nf
+++ b/modules/nf-core/salsa2/main.nf
@@ -49,4 +49,20 @@ process SALSA2 {
         SALSA2: $VERSION
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def VERSION = '2.3' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    """
+    touch ${prefix}_scaffolds_FINAL.fasta
+    touch ${prefix}_scaffolds_FINAL.agp
+    
+    mkdir -p salsa2_output
+    touch salsa2_output/${prefix}_scaffolds_FINAL.original-coordinates.agp
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        SALSA2: $VERSION
+    END_VERSIONS
+    """
 }
diff --git a/modules/nf-core/salsa2/tests/main.nf.test b/modules/nf-core/salsa2/tests/main.nf.test
index 6897f23..3426a3e 100644
--- a/modules/nf-core/salsa2/tests/main.nf.test
+++ b/modules/nf-core/salsa2/tests/main.nf.test
@@ -34,6 +34,34 @@ nextflow_process {
                 { assert snapshot(process.out.agp).match("agp") },
                  { assert snapshot(process.out.versions).match("versions") }
             )
+    }
+
+    test("Single-Read - stub") {
+        options "-stub"
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test', single_end: false ], // meta map
+                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true),
+                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta.fai', checkIfExists: true)
+                ]
+                input[1] = [
+                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/bed/test.bed', checkIfExists: true)
+                ]
+                input[2] = []
+                input[3] = []
+                input[4] = []
+                """
+            }
+        }
+
+        then {
+            assertAll (
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
         }
     }
 }
diff --git a/modules/nf-core/shasta/main.nf b/modules/nf-core/shasta/main.nf
index b55df9f..e2e07c2 100644
--- a/modules/nf-core/shasta/main.nf
+++ b/modules/nf-core/shasta/main.nf
@@ -46,4 +46,17 @@ process SHASTA {
         shasta: \$(shasta --version | head -n 1 | cut -f 3 -d " ")
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    echo "" | gzip > ${prefix}_Assembly.fasta.gz
+    echo "" | gzip > ${prefix}_Assembly.gfa.gz
+    mkdir ShastaRun
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        shasta: \$(shasta --version | head -n 1 | cut -f 3 -d " ")
+    END_VERSIONS
+    """
 }
diff --git a/modules/nf-core/shasta/tests/main.nf.test b/modules/nf-core/shasta/tests/main.nf.test
index ee66549..b45c625 100644
--- a/modules/nf-core/shasta/tests/main.nf.test
+++ b/modules/nf-core/shasta/tests/main.nf.test
@@ -38,4 +38,28 @@ nextflow_process {
         }
     }
 
+    test("test-shasta - stub") {
+
+        options "-stub"
+
+        when {
+            process {
+                """
+                input[0] = [
+				    [ id:'test', model:'Nanopore-Dec2019' ], // meta map
+				    [ file(params.modules_testdata_base_path + 'genomics/prokaryotes/candidatus_portiera_aleyrodidarum/nanopore/fastq/test.fastq.gz', checkIfExists: true) ],
+				]
+
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+    }
+
 }
diff --git a/modules/nf-core/shovill/main.nf b/modules/nf-core/shovill/main.nf
index c54c68f..400da4d 100644
--- a/modules/nf-core/shovill/main.nf
+++ b/modules/nf-core/shovill/main.nf
@@ -34,4 +34,14 @@ process SHOVILL {
         --outdir ./ \\
         --force
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch contigs.fa
+    touch shovill.corrections
+    touch shovill.log
+    touch spades.fasta
+    touch contigs.fastg
+    """
 }
diff --git a/modules/nf-core/shovill/tests/main.nf.test b/modules/nf-core/shovill/tests/main.nf.test
index 99a1e22..a27353d 100644
--- a/modules/nf-core/shovill/tests/main.nf.test
+++ b/modules/nf-core/shovill/tests/main.nf.test
@@ -115,4 +115,26 @@ nextflow_process {
         }
 
     }
+
+    test("shovill - stub") {
+        options "-stub"
+        when {
+            process {
+                """
+                input[0] = [ [ id:'test', single_end:false ], // meta map
+                [ file("https://github.com/nf-core/test-datasets/raw/bacass/ERR044595_1M_1.fastq.gz", checkIfExists: true),
+                  file("https://github.com/nf-core/test-datasets/raw/bacass/ERR044595_1M_2.fastq.gz", checkIfExists: true) ]
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+
+    }
 }
