diff --git a/modules/nf-core/fastqscan/main.nf b/modules/nf-core/fastqscan/main.nf
index ac5c13fbc4..9385fd5f87 100644
--- a/modules/nf-core/fastqscan/main.nf
+++ b/modules/nf-core/fastqscan/main.nf
@@ -30,4 +30,15 @@ process FASTQSCAN {
         fastqscan: \$( echo \$(fastq-scan -v 2>&1) | sed 's/^.*fastq-scan //' )
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${prefix}.json
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        fastqscan: \$( echo \$(fastq-scan -v 2>&1) | sed 's/^.*fastq-scan //' )
+    END_VERSIONS
+    """
 }
diff --git a/modules/nf-core/fastqscan/tests/main.nf.test b/modules/nf-core/fastqscan/tests/main.nf.test
index aa08cdb690..64eb8dc705 100644
--- a/modules/nf-core/fastqscan/tests/main.nf.test
+++ b/modules/nf-core/fastqscan/tests/main.nf.test
@@ -32,4 +32,31 @@ nextflow_process {
 
     }
 
+    test("Should run without failures - stub") {
+
+        options "-stub"
+
+        when {
+            params {
+                outdir = "$outputDir"
+            }
+            process {
+                """
+                input[0] =  [
+                                [ id:'test', single_end:true ], // meta map
+                                file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)
+                            ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+
+    }
+
 }
diff --git a/modules/nf-core/filtlong/main.nf b/modules/nf-core/filtlong/main.nf
index 065397458c..cd8a45825c 100644
--- a/modules/nf-core/filtlong/main.nf
+++ b/modules/nf-core/filtlong/main.nf
@@ -31,4 +31,12 @@ process FILTLONG {
         2>| >(tee ${prefix}.log >&2) \\
         | gzip -n > ${prefix}.fastq.gz
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    if ("$longreads" == "${prefix}.fastq.gz") error "Longread FASTQ input and output names are the same, set prefix in module configuration to disambiguate!"
+    """
+    echo "" | gzip > ${prefix}.fastq.gz
+    touch ${prefix}.log
+    """
 }
diff --git a/modules/nf-core/filtlong/tests/main.nf.test b/modules/nf-core/filtlong/tests/main.nf.test
index 044fa337e3..9ce475fe4b 100644
--- a/modules/nf-core/filtlong/tests/main.nf.test
+++ b/modules/nf-core/filtlong/tests/main.nf.test
@@ -93,4 +93,32 @@ nextflow_process {
         }
 
     }
+
+    test("sarscov2 nanopore [fastq] - stub") {
+
+        options "-stub"
+
+        when {
+            params {
+                outdir = "$outputDir"
+            }
+            process {
+                """
+                input[0] = [
+                    [ id:'test', single_end:false ], // meta map
+                    [],
+                    [ file(params.modules_testdata_base_path + 'genomics/sarscov2/nanopore/fastq/test.fastq.gz', checkIfExists: true) ]
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+
+    }
 }
diff --git a/modules/nf-core/prinseqplusplus/main.nf b/modules/nf-core/prinseqplusplus/main.nf
index 403a51bc84..1649194eda 100644
--- a/modules/nf-core/prinseqplusplus/main.nf
+++ b/modules/nf-core/prinseqplusplus/main.nf
@@ -58,4 +58,34 @@ process PRINSEQPLUSPLUS {
         END_VERSIONS
         """
     }
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    if (meta.single_end) {
+        """
+        echo "" | gzip > ${prefix}_good_out.fastq.gz
+        echo "" | gzip > ${prefix}_bad_out.fastq.gz
+        touch ${prefix}.log
+
+        cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            prinseqplusplus: \$(echo \$(prinseq++ --version | cut -f 2 -d ' ' ))
+        END_VERSIONS
+        """
+    } else {
+        """
+        echo "" | gzip > ${prefix}_good_out_R1.fastq.gz
+        echo "" | gzip > ${prefix}_good_out_R2.fastq.gz
+        echo "" | gzip > ${prefix}_single_out_R1.fastq.gz
+        echo "" | gzip > ${prefix}_single_out_R2.fastq.gz
+        echo "" | gzip > ${prefix}_bad_out_R1.fastq.gz
+        echo "" | gzip > ${prefix}_bad_out_R2.fastq.gz
+        touch ${prefix}.log
+
+        cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            prinseqplusplus: \$(echo \$(prinseq++ --version | cut -f 2 -d ' ' ))
+        END_VERSIONS
+        """
+    }
 }
diff --git a/modules/nf-core/prinseqplusplus/tests/main.nf.test b/modules/nf-core/prinseqplusplus/tests/main.nf.test
index a6708ce96a..30d20849e3 100644
--- a/modules/nf-core/prinseqplusplus/tests/main.nf.test
+++ b/modules/nf-core/prinseqplusplus/tests/main.nf.test
@@ -71,4 +71,66 @@ nextflow_process {
         }
     }
 
+    test("test-prinseqplusplus-single-end - stub") {
+
+        options "-stub"
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test', single_end:true ], // meta map
+                    [ file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true) ]
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                    file(process.out.good_reads[0][1]).name, // unstable
+                    process.out.single_reads,
+                    process.out.bad_reads,
+                    file(process.out.log[0][1]).name,
+                    process.out.versions
+                    ).match()
+                }
+            )
+        }
+    }
+
+    test("test-prinseqplusplus-paired-end - stub") {
+
+        options "-stub"
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test', single_end:false ], // meta map
+                    [
+                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
+                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
+                    ]
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                    process.out.good_reads[0][1].collect { file(it).name }, // unstable
+                    process.out.single_reads[0][1].collect { file(it).name }, // empty: d41d8cd98f00b204e9800998ecf8427e
+                    process.out.bad_reads,
+                    file(process.out.log[0][1]).name,
+                    process.out.versions
+                    ).match()
+                }
+            )
+        }
+    }
 }
diff --git a/modules/nf-core/pycoqc/main.nf b/modules/nf-core/pycoqc/main.nf
index e748aac136..1339eff68d 100644
--- a/modules/nf-core/pycoqc/main.nf
+++ b/modules/nf-core/pycoqc/main.nf
@@ -33,4 +33,16 @@ process PYCOQC {
         pycoqc: \$(pycoQC --version 2>&1 | sed 's/^.*pycoQC v//; s/ .*\$//')
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${prefix}.html
+    touch ${prefix}.json
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        pycoqc: \$(pycoQC --version 2>&1 | sed 's/^.*pycoQC v//; s/ .*\$//')
+    END_VERSIONS
+    """
 }
diff --git a/modules/nf-core/pycoqc/tests/main.nf.test b/modules/nf-core/pycoqc/tests/main.nf.test
index 0df42e226b..cb1ec8fdab 100644
--- a/modules/nf-core/pycoqc/tests/main.nf.test
+++ b/modules/nf-core/pycoqc/tests/main.nf.test
@@ -40,4 +40,32 @@ nextflow_process {
 
     }
 
+    test("sarscov2 nanopore [sequencing_summary.txt] - stub") {
+
+        options "-stub"
+
+        when {
+            params {
+                outdir = "$outputDir"
+            }
+            process {
+                """
+                input[0] =  [
+                                [ id:'test' ], // meta map
+                                file(params.modules_testdata_base_path + 'genomics/sarscov2/nanopore/sequencing_summary/test.sequencing_summary.txt', checkIfExists: true)
+                            ]
+
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+
+    }
+
 }
diff --git a/modules/nf-core/rasusa/main.nf b/modules/nf-core/rasusa/main.nf
index a1d36debd5..aa900fa2bb 100644
--- a/modules/nf-core/rasusa/main.nf
+++ b/modules/nf-core/rasusa/main.nf
@@ -30,4 +30,17 @@ process RASUSA {
         --input $reads \\
         $output
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    if (meta.single_end) {
+        """
+        echo "" | gzip > ${prefix}.fastq.gz
+        """
+    } else {
+        """
+        echo "" | gzip > ${prefix}_1.fastq.gz
+        echo "" | gzip > ${prefix}_2.fastq.gz
+        """
+    }
 }
diff --git a/modules/nf-core/rasusa/tests/main.nf.test b/modules/nf-core/rasusa/tests/main.nf.test
index 90ea7fc75d..35b3526552 100644
--- a/modules/nf-core/rasusa/tests/main.nf.test
+++ b/modules/nf-core/rasusa/tests/main.nf.test
@@ -35,4 +35,34 @@ nextflow_process {
 
     }
 
+    test("Should run without failures - stub") {
+
+        options "-stub"
+
+        when {
+            params {
+                outdir = "$outputDir"
+            }
+            process {
+                """
+                input[0] =  [ [ id:'testfile', single_end:false], // meta map
+                                [ file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
+                                  file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
+                                ],
+                                "1000000b"
+                            ]
+                input[1] = 100
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+
+    }
+
 }
