--- a/modules/nf-core/gatk4/haplotypecaller/main.nf
+++ b/modules/nf-core/gatk4/haplotypecaller/main.nf
@@ -61,8 +61,8 @@
 
     def stub_realigned_bam = bamout_command ? "touch ${prefix.replaceAll('.g\\s*$', '')}.realigned.bam" : ""
     """
-    touch ${prefix}.vcf.gz
-    touch ${prefix}.vcf.gz.tbi
+    echo '' | gzip > ${prefix}.vcf.gz
+    echo '' | gzip > ${prefix}.vcf.gz.tbi
     ${stub_realigned_bam}
     """
 }
